README.md

```diff
@@ -186,7 +186,25 @@ Contributing
 ---
 
 When adding or modifying a query in this repository, make your changes in the
-`sql/` directory. 
+`sql/` directory.
+
+When adding a new library to the Python requirements, first compile the
+dependencies into the constraints, and then add the library to the requirements.
+
+```bash
+# If not installed:
+pip install hashin pip-tools
+
+# Add the dependency to constraints.txt e.g. jinja2. The purpose of constraints.txt
+# is to avoid cluttering the real dependencies requirements.txt.
+hashin jinja2==2.11.1 -r constraints.txt
+
+# Compile constraints to obtain hashes for dependencies
+pip-compile --generate-hashes constraints.txt
+
+# Finally ass dependency to requirements.txt
+hashin jinja2==2.11.1
+```
```

Comment:
@relud Does this piece of documentation sound?

Reply:
unfortunately no. The correct steps are:
```bash
# create and activate a python virtual environment
python3 -m venv venv/
. venv/bin/activate

# If not installed:
pip install hashin

# Add the dependency to requirements.txt e.g. Jinja2.
hashin Jinja2==2.11.1

# Use pip install to check for missing meta-dependencies
pip install --require-hashes -r requirements.txt

# Add hashes for missing meta-dependencies until pip install succeeds
hashin MarkupSafe -r constraints.txt

# deactivate the python virtual environment
deactivate
```

note that requirements go in *either* `constraints.txt` or `requirements.txt`, but not both.

Reply:
Okay, thanks. It looks like this command was what I was missing when running into missing transitive dependencies:

```
# Use pip install to check for missing meta-dependencies
pip install --require-hashes -r requirements.txt
```

Reply:
filed #736 to start using `pip-compile`, so that this is easier in the future.